src/components/interaction-designer/block-editors/GroupMembershipEditor.vue

```diff
@@ -1,153 +1,204 @@
-            :valid-state="isValid" />
-        </div>
-      </validation-message>
-
-      <validation-message
-        v-if="membershipAction === MEMBERSHIP_ACTION.SET_FROM_EXPRESSION"
-        #input-control="{ isValid }"
-        :message-key="`block/${block.uuid}/config/is_member`">
-        <expression-input
-          :label="'flow-builder.value-expression' | trans"
-          :placeholder="'flow-builder.enter-expression' | trans"
-          :label-class="'font-weight-bold'"
-          :current-expression="isMember"
-          :valid-state="isValid"
-          @commitExpressionChange="updateIsMemberExpression" />
-      </validation-message>
+      <vue-multiselect
+        v-if="isGroupListVisible"
+        v-model="selectedGroups"
+        :internal-search="!hasCustomSearch"
+        :is-loading="groupsLoading"
+        :options="groupOptions"
+        :taggable="!hasCustomSearch"
+        :multiple="true"
+        track-by="group_key"
+        label="group_name"
+        @seach-change="onSearchChange"
+        @tag="onGroupAdd" />
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import {IBlock} from '@floip/flow-runner'
+import {IBlock, ISetGroupMembershipBlockConfig, IGroupMembership} from '@floip/flow-runner'
 import {Component, Prop} from 'vue-property-decorator'
 import {namespace} from 'vuex-class'
+import VueMultiselect from 'vue-multiselect'
 import Lang from '@/lib/filters/lang'
-import {get} from 'lodash'
 import {mixins} from 'vue-class-component'
 
 const flowVuexNamespace = namespace('flow')
 
-const NULL_STRING_EXPRESSION = '@(null)'
-const TRUTHY_STRING_EXPRESSION = '@(true)'
-const EMPTY_STRING_EXPRESSION = ''
+type MembershipAction = {
+  id: string,
+  label: string,
+  value: string,
+}
+
+enum MEMBERSHIP_ACTION {
+  ADD = 'add',
+  REMOVE = 'remove',
+  CLEAR = 'clear',
+}
 
-@Component({})
+@Component({
+  components: {VueMultiselect},
+})
 export class GroupMembershipEditor extends mixins(Lang) {
   @Prop() readonly block!: IBlock
 
-  MEMBERSHIP_ACTION = {
-    ADD: 'add',
-    SET_FROM_EXPRESSION: 'set_from_expression',
-    REMOVE: 'remove',
+  /*
+   * The following two props should be used together
+   * when a user is only be allowed to select groups
+   * from a pre-defined list.
+   *
+   * availableGroups = [
+   *  { group_key: 'gr1', group_name: 'Foo' },
+   *  { group_key: 'gr2', group_name: 'Bar' },
+   * ]
+   *
+   * groupsLoading should be set when the availableGroups
+   * list is unknown yet, e.g., while an asynchronous request
+   * to fetch the list after the group-search event
+   * has not been resolved.
+   */
+  @Prop() readonly availableGroups?: IGroupMembership[]
+  @Prop({type: Boolean, default: false}) readonly groupsLoading!: boolean
+
+  customGroupOptions: IGroupMembership[] = []
+  cachedGroupsSelection: IGroupMembership[] = []
+
+  get hasCustomSearch(): boolean {
+    return this.availableGroups !== undefined
```

Comment:
@object-required , this wasn't really clarified in the design & ticket, sorry.
I think, we still need the `search` feature even with a pre-defined list, not only for community version. In fact, in voto5, we might have a long list of groups.

- [ ] allow the search feature for all versions